mod/groups/actions/groups/edit.php

```diff
@@ -94,7 +94,7 @@
 }
 
 // Set group tool options
-$tool_options = elgg_get_config('group_tool_options');
+$tool_options = groups_get_group_tool_options($group);
```

Comment:
I pass around unsaved entities on some occasions. I think the problem here with hook calling `save()` assuming that entity already exists (which they shouldn't).